hugegraph-core/src/main/java/com/baidu/hugegraph/backend/serializer/BinarySerializer.java

```diff
@@ -570,10 +575,39 @@ public BackendEntry writeId(HugeType type, Id id) {
     @Override
     protected Id writeQueryId(HugeType type, Id id) {
         if (type.isEdge()) {
-            id = writeEdgeId(id);
+            if(this.enablePartition){
+                EdgeId edgeId;
+                if (id instanceof EdgeId) {
+                    edgeId = (EdgeId) id;
+                } else {
+                    edgeId = EdgeId.parse(id.asString());
+                }
+
+                BytesBuffer bf = BytesBuffer.allocate(BytesBuffer.BUF_EDGE_ID);
+                String startNode = com.baidu.hugegraph.util.Bytes.toHex(edgeId.ownerVertexId().asBytes());
```

Comment:
Bytes.toHex is an expensive operation, please use getPartition(Id id), then calculate hash code through id.asBytes() directly